### ISSUE TITLE
How to prevent Program.cs running with Heroku deploy

### ISSUE THREAD

**Issue by Rhonage** [jincod/dotnetcore-buildpack#130]
All in the title really. Sorry if this is a newb question, I'm still figuring out how Heroku and .NET Core work together.

I see that Heroku executes the following when deployed.

`$ cd /app/heroku_output && ./My.Project.Name`

Is there a way to:

1. Stop this happening for each deploy?
2. Specify a particular class or method to run?

Thanks in advance.

**Comment by jincod** [comment]
> I see that Heroku executes the following when deployed.
> $ cd /app/heroku_output && ./My.Project.Name

No, it doesn't execute this during deploy. Full snippet:

https://github.com/jincod/dotnetcore-buildpack/blob/8d57e337893ce99df690cd8140996bee1a78154f/bin/compile#L76-L78

More info about Heroku: [How Heroku Works](https://devcenter.heroku.com/articles/how-heroku-works)

**Comment by Rhonage** [comment]
@jincod Thanks for that. Each time I push my repo to Heroku though, it runs Program.cs (my repo is a console app, which will be scheduled to run every hour - so not a web server).

Is this something that can be changed by setting up a custom Profile at the room of my repo? I more or less want to do something like this (pseudocode):

```
...
if [ -f ${BUILD_DIR}/Procfile ]; then
	topic "WARNING"
	echo "Be careful with custom Procfile" | indent
else
	echo "Add web process to Procfile" | indent
	cat << EOT >> ${BUILD_DIR}/Procfile

# Have some sort of environment variable where DEPLOY_ON_PUSH can be set to false.
if [DEPLOY_ON_PUSH]; then
        web: cd \$HOME/heroku_output && ./${PROJECT_NAME}
```

Thanks in advance for your awesome responses.

**Comment by jincod** [comment]
You need a custom Procfile

Check this issue https://github.com/jincod/dotnetcore-buildpack/issues/116 and Heroku article https://devcenter.heroku.com/articles/process-model

**Comment by Rhonage** [comment]
@jincod  Thanks again for your fast reply.

I have added the following to a custom Procfile for my "Haven.Sender" repo:

`worker: cd $HOME/heroku_output && ./Haven.Sender`

But strangely it looks like the build didn't detect the custom Procfile, and has concatenated it to my custom one.

![image](https://user-images.githubusercontent.com/5068726/98773527-f9fb2680-244d-11eb-9492-9fc4ba472354.png)

My project layout is as follows:
```
- Haven.Sender (contains .git, and Procfile)
--- Haven.Sender (contains the actual Haven.Sender.csproj, bin/obj files etc)
--- Shared (reference projects)

```
Have I done something wrong?

**Comment by jincod** [comment]
Make sure you use the latest version of buildpack. 

It has been fixed already https://github.com/jincod/dotnetcore-buildpack/commit/ccbbc7109657b9f5bded55e3bdf8174ee0094d8a#diff-a64a7d9b82578ca3053416d56631a61e0ceabed87619415663030e3842429e05

**Comment by Rhonage** [comment]
Yup, I'm using the latest. I even removed the buildpack from Heroku, and added it again.

I just did another deploy, and here's the text. Unfortunate it also ran when deployed, so it sounds like it's not detecting the Procfile for some reason. Any ideas? Not a huge issue really, I just have to be careful when deploying.

remote:   Haven.Data -> /tmp/build_a68c3247/Shared/Haven.Data/bin/Release/netcoreapp3.1/Haven.Data.dll        
remote:   Haven.Sender -> /tmp/build_a68c3247/Haven.Sender/bin/Release/netcoreapp3.1/linux-x64/Haven.Sender.dll        
remote:   Haven.Sender -> /tmp/build_a68c3247/heroku_output/        
**remote:        Add web process to Procfile**        
remote: -----> Discovering process types        
**remote:        Procfile declares types -> worker**        
remote: 
remote: -----> Compressing...        


remote:        Done: 98.2M        
remote: -----> Launching...        
remote:        Released v17        
remote:        https://novacode-haven-sender.herokuapp.com/ deployed to Heroku        
remote: 

remote: Verifying deploy... done.        
Pushing to https://git.heroku.com/novacode-haven-sender.git
To https://git.heroku.com/novacode-haven-sender.git
   2044604..d552827  master -> master
updating local tracking ref 'refs/remotes/heroku/master'


Completed successfully.
